vignettes/available-sass-variables.Rmd

```diff
@@ -0,0 +1,251 @@
+    # map-get function is used to reference a map built earlier so make sure not
+    # to link to it as it's not available in the table
+    value = str_replace_all(
+      value,
+      "(?<!map-get\\()(\\$[\\w|-]+)",
+      "<span class='dep-link'>\\1</span>"
+    ),
+    section = glue("Section: {section} <a href=\"{section_link}\"><i class=\"fas fa-external-link-alt\"></i></a>"),
+    declaration = glue("<a href=\"{link_to_pos_in_script(line_number)}\"><i class=\"fas fa-external-link-alt\"></i></a>")
+  ) %>%
+  filter(!value_is_function) %>%
+  select(
+    section,
+    variable,
+    value,
+    comment,
+    declaration
+  ) -> variables_df 
+```
+
+
+## Modifying Bootstrap Sass variable values
+
+Bootstrap uses `sass` to generate its final CSS. While a large portion of the customization you may want to do to your theme is easily done with the common arguments in `bs_theme()`, sometimes you may want finer-grain control. To enable this, `bslib` let's you change any of the variables used in the Bootstrap `sass` code. For instance, if you want to adjust the padding amount after each paragraph, that is available with the variable `paragraph-margin-bottom`, if we wanted to double the default amount (`1rem`) we could do so by simply passing the name of the variable to our `bs_theme()` call...
+
+```{r, eval = FALSE, echo = TRUE}
+theme <- bs_theme(
+  ...,
+  # Double space given after each paragraph
+  "paragraph-margin-bottom" = "2rem"
+)
+```
+
+
+### What's the "`$`" doing in front of the variables?
+You may notice in the table below that the variables are prefixed with a `$` this is a `sass` convention and you should omit the `$` in your calls to `bs_theme()`. However, if you wanted to reference another variable in your declaration of a value, such as making the `dark` variable equal to `grey-900` instead of `grey-800`, you _would_ need to use the `$` prefix when referring to the dependency. Also, you will need to tell `bslib` to place its definition later than `grey-900`'s with `bs_add_variables(..., .where = "declarations")`:
+
+```{r, eval = FALSE, echo = TRUE}
+theme <- bs_theme(
+  ...,
+  "paragraph-margin-bottom" = "2rem"
+) %>% 
+  # When referring to other variables, prefix with dollar-sign
+  bs_add_variables("dark" = "$gray-900", .where = "declarations")
+```
+
+For more details on this see the `sass` variables section of the [theming article.](https://rstudio.github.io/bslib/articles/theming.html#sass-variables-1)
+
+
+## All customizable variables
+
+There are a _lot_ of these `sass` variables available for adjusting. Below is a table that shows all of them, along with basic comments on variables functionality as provided in the original Bootstrap code. For more details on how customization of these variables works, see the function `bs_add_variables()`.
+
+_Note: these variables are specifically for Bootstrap 4 and many won't work for Bootstrap 3._
```

Comment:
This should be more prominent (put above or in the table?)

Reply:
Specifically the "BS4 only" note?

Reply:
Oh, yea, just the note. Also, referencing the R docs above, could point to https://rstudio.github.io/bslib/articles/theming.html#sass-variables-1 instead

Reply:
Relatedly, we should probably link to this table towards the end of that section

Reply:
Would also be good to replace existing hyperlinks in the docs to point to the table instead, for example https://github.com/rstudio/bslib/blame/master/README.md#L61

Reply:
I'll addressing these things throughout the day